src/binary/ibc-setup/commands/init.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import path from 'path';
+
+import { Command } from 'commander';
+import yaml from 'js-yaml';
+
+import { GlobalOptions } from '../types';
+import { resolveOption } from '../utils/resolve-option';
+
+type Options = GlobalOptions & {
+  src: string;
+  dest: string;
+};
+
+type ProgramOptions = Partial<Options>;
+
+function getDefaultHome() {
+  if (!process.env.HOME) {
+    throw new Error('$HOME environment variable is not set.');
+  }
+
+  return `${process.env.HOME}/.ibc-setup`;
+}
+
+const REGISTRY_FILE = 'registry.yaml';
+
+export function init(_: unknown, program: Command) {
+  const programOptions: ProgramOptions = program.opts();
+
+  const options = {
+    src: resolveOption('src', programOptions.src, process.env.RELAYER_SRC),
+    dest: resolveOption('dest', programOptions.dest, process.env.RELAYER_DEST),
+    home: resolveOption('home', programOptions.home, getDefaultHome),
+  };
+
+  run(options);
+}
+
+export function run(options: Options) {
+  if (!fs.existsSync(options.home)) {
+    fs.mkdirSync(options.home, { recursive: true });
+    console.log(`Initialized home directory at ${options.home}`);
+  } else if (!fs.lstatSync(options.home).isDirectory()) {
+    throw new Error(`${options.home} must be a directory. It is a file.`);
+  }
+
+  const REGISTRY_FILE_PATH = path.join(options.home, REGISTRY_FILE);
+  if (!fs.existsSync(REGISTRY_FILE_PATH)) {
+    // TODO: download registry.yaml from default location
```

Comment:
I guess we want to download this from some remote registry and **not** ship with the binary, right?

Maybe for now, we can pull it from GitHub?

Reply:
Yes. Use this link: https://raw.githubusercontent.com/confio/ts-relayer/main/demo/registry.yaml

We can change that later, but it works for now.

Reply:
https://github.com/confio/ts-relayer/pull/73/commits/85e0baa59a1575ce930cfe4fa1d3d2fa91d25ddc